commit a3e5efd845589c11f93935c0a0b13262b1e912b3
Author: Ulrich Drepper <drepper@redhat.com>
Date:   Tue Dec 9 23:04:03 2008 +0000

    (pthread_cleanup_pop): Use { } as empty loop body instead of ; to avoid gcc
    warnings.
    (pthread_cleanup_pop_restore_np): Likewise.

---
 nptl/sysdeps/pthread/pthread.h |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/nptl/sysdeps/pthread/pthread.h
+++ b/nptl/sysdeps/pthread/pthread.h
@@ -655,7 +655,7 @@
 /* Remove a cleanup handler installed by the matching pthread_cleanup_push.
    If EXECUTE is non-zero, the handler function is called. */
 # define pthread_cleanup_pop(execute) \
-      do; while (0); /* Empty to allow label before pthread_cleanup_pop.  */  \
+      do { } while (0);/* Empty to allow label before pthread_cleanup_pop.  */\
     } while (0);							      \
     __pthread_unregister_cancel (&__cancel_buf);			      \
     if (execute)							      \
@@ -691,7 +691,7 @@
    restores the cancellation type that was in effect when the matching
    pthread_cleanup_push_defer was called.  */
 #  define pthread_cleanup_pop_restore_np(execute) \
-      do; while (0); /* Empty to allow label before pthread_cleanup_pop.  */  \
+      do { } while (0);/* Empty to allow label before pthread_cleanup_pop.  */\
     } while (0);							      \
     __pthread_unregister_cancel_restore (&__cancel_buf);		      \
     if (execute)							      \
